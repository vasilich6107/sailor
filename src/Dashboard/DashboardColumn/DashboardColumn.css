.dashboard__column  {
    display: flex;
    flex-direction: column;
    flex-grow: 1;
    flex-basis: 0;
}

.dashboard__column-title  {
    font-weight: bold;
}

.dashboard__column-content  {
    display: flex;
    align-items: center;
    flex-direction: column;
}