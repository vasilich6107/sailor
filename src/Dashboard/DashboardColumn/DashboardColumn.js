// @flow
import React from 'react';
import DashboardCell from '../DashboardCell';
import type { Applicant } from '../../types/Applicant';

import './DashboardColumn.css';

type Props = {
  columnName: number,
  applicants: Array<Applicant>,
  onNext: (uuid: string) => void,
  onPrevious: (uuid: string) => void,
};

const DashboardColumn = ({ columnName, applicants, ...props }: Props) => (
  <div className="dashboard__column">
    <div className="dashboard__column-title">{columnName}</div>
    <div className="dashboard__column-content">
      {applicants.map(applicant => (
        <DashboardCell key={applicant.login.uuid} applicant={applicant} {...props} />
      ))}
    </div>
  </div>
);

export default DashboardColumn;
