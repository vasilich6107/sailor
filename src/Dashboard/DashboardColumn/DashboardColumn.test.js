import React from 'react';
import { shallow } from 'enzyme';
import DashboardColumn from './DashboardColumn';

jest.mock('../DashboardCell', () => 'DashboardCell');

const applicants = [
  {
    login: {
      uuid: '5d204000-387b-4741-8ba9-7c121ae6adab',
    },
  },
  {
    login: {
      uuid: '40fdd877-b7f1-47bd-8c6b-8b2129bbf3fd',
    },
  },
  {
    login: {
      uuid: '674ed62e-a5f2-42a1-9ba8-2c00c3671801',
    },
  },
];

const columnName = 'Test name';

it('Renders dashboard column', () => {
  expect(shallow(<DashboardColumn applicants={applicants} columnName={columnName} />)).toMatchSnapshot();
});

it('Renders empty dashboard column', () => {
  expect(shallow(<DashboardColumn applicants={[]} columnName={columnName} />)).toMatchSnapshot();
});
