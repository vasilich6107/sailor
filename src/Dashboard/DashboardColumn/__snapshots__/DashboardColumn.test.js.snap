// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders dashboard column 1`] = `
<div
  className="dashboard__column"
>
  <div
    className="dashboard__column-title"
  >
    Test name
  </div>
  <div
    className="dashboard__column-content"
  >
    <DashboardCell
      applicant={
        Object {
          "login": Object {
            "uuid": "5d204000-387b-4741-8ba9-7c121ae6adab",
          },
        }
      }
      key="5d204000-387b-4741-8ba9-7c121ae6adab"
    />
    <DashboardCell
      applicant={
        Object {
          "login": Object {
            "uuid": "40fdd877-b7f1-47bd-8c6b-8b2129bbf3fd",
          },
        }
      }
      key="40fdd877-b7f1-47bd-8c6b-8b2129bbf3fd"
    />
    <DashboardCell
      applicant={
        Object {
          "login": Object {
            "uuid": "674ed62e-a5f2-42a1-9ba8-2c00c3671801",
          },
        }
      }
      key="674ed62e-a5f2-42a1-9ba8-2c00c3671801"
    />
  </div>
</div>
`;

exports[`Renders empty dashboard column 1`] = `
<div
  className="dashboard__column"
>
  <div
    className="dashboard__column-title"
  >
    Test name
  </div>
  <div
    className="dashboard__column-content"
  />
</div>
`;
